pkg/ensurance/opamock/opamock.go

```diff
@@ -0,0 +1,12 @@
+package opamock
+
+func OpaEval(metricName string, targetValue float64, value float64) (bool, error) {
```

Comment:
Here the function interface should be something like Eval(ctx, input).
metricName, targetValue... all these kind of things should not be placed here.

Reply:
type Logic interface {
	EvalWithMetric(metricName string, targetValue float64, value float64) (bool, error)
	EvalWithRaw(input string, rule string) (bool, error)
}

I modified it like this, please have a look  it feasible now ?